scripts/eval/eval.py

```diff
@@ -287,6 +336,15 @@ def main(cfg: DictConfig) -> Tuple[List[Trainer], pd.DataFrame]:
     models_df = None
     composite_scores = None
     trainers = []
+
+    mosaicml_logger = find_mosaicml_logger([
+        build_logger(name, logger_cfg)
```

Comment:
@aspfohl mentioned that since `evaluate_model` is called twice, we're likely to overwrite some values with our current logging paradigms. I think that logging everything at once is good from an analytics perspective because we can see everything before all of the calls to `evaluate_model` are ran (although i'm not sure how long this usually takes?).

the reason why i built the loggers again on the outside is that we [pop/update the `loggers_cfg` metadata every time `evaluate_model` is called](https://github.com/mosaicml/llm-foundry/blob/angel/log-data-for-run-analytics/scripts/eval/eval.py#L164). (naive question) why do we do this, and how should we go about this?

I have two ideas: 
1. move all analytics logging into `main`. for `loggers_cfg`, can we pop/update the metadata once as well? If not, we can pass in a function (or `loggers_cfg`) to perform these operations in `evaluate_model`. **this is what I will do right now**
2. if logging for every model is better, we can log analytics for each one, but rename them or place them in a larger list so that we don't have naming conflicts/overwrites in our metadata. For example, we can create a list in metadata, `llmfoundry/models`, and update this list with model data every time `evaluate_model` is called.

Thoughts? @dakinggg @irenedea @aspfohl

Reply:
Oh i see, yeah that makes sense. Let's just build all the loggers once and then pass the built loggers to evaluate_model